### PR TITLE
use Ibexa

### PR DESCRIPTION
**Pull request by RemyNovactive**
| Q             | A
| ------------- | ---
| Branch?       | feat-menumanager-ibx4-use-Ibexa => feat-menumanager-ibx4
| Bug fix?      | yes
| New feature?  | no 
| BC breaks?    | no
| Fixed tickets | [#114982 - [LEF-Recette] Corriger l'édition du menu : ezmenumanagerbundle](https://almaviacx.easyredmine.com/issues/114982)

replace `use ez` by `use Ibexa` 

!! Symfony Exception **TypeError** HTTP 500 Internal Server Error

ReflectionProperty::getValue(): Argument #1 ($object) must be provided for instance properties

### CODE REVIEW

File: components/MenuManagerBundle/src/bundle/Controller/AdminController.php [hunk 1]
```diff
@@ -158,10 +153,13 @@ public function newAction(Request $request)
...
      *
-     * @return \Symfony\Component\HttpFoundation\RedirectResponse|\Symfony\Component\HttpFoundation\Response
+     * @param Request $request
+     * @param Menu $menu
+     * @return RedirectResponse|Response
      */
-    public function editAction(Request $request, Menu $menu)
+    public function editAction(Request $request, Menu $menu): RedirectResponse|Response
     {
+        // dump($menu); // Ici on a bien un Novactive\EzMenuManagerBundle\Entity\Menu
```
Comment (RemyNovactive) [inline components/MenuManagerBundle/src/bundle/Controller/AdminController.php:162]: // dump($menu); // Ici on a bien un Novactive\EzMenuManagerBundle\Entity\Menu

File: components/MenuManagerBundle/src/lib/FieldType/MenuItem/Value.php [hunk 2]
```diff
@@ -22,23 +22,24 @@ class Value extends BaseValue
...
+        parent::__construct();
     }
 
     /**
      * @see \eZ\Publish\Core\FieldType\Value
      */
     public function __toString()
     {
-        return (string) $this->menuItems;
+        return (string) $this->menuItems; // TODO Array to string conversion
```
Comment (RemyNovactive) [inline components/MenuManagerBundle/src/lib/FieldType/MenuItem/Value.php:43]: TODO Array to string conversion

File: components/MenuManagerBundle/src/lib/Service/DataTransformer/MenuItemsCollectionTransformer.php [hunk 3]
```diff
@@ -32,10 +32,13 @@ public function __construct(MenuItemConverter $menuItemConverter)
...
      *
-     * @return array|null the value's hash, or null if $value was not a FieldType Value
+     * @return string|false the value's hash, or null if $value was not a FieldType Value
+     * @throws UnexpectedTypeException
      */
-    public function transform($value)
+    public function transform($value): bool|string
     {
+        dump($value); // ici on a un Doctrine\ORM\PersistentCollection
+        dump($value->getValues()); // ERROR 500 TypeError ReflectionProperty::getValue(): Argument #1 ($object) must be provided for instance properties
```
Comment (RemyNovactive) [inline components/MenuManagerBundle/src/lib/Service/DataTransformer/MenuItemsCollectionTransformer.php:41]: dump($value); // ici on a un Doctrine\ORM\PersistentCollection
        dump($value->getValues()); // ERROR 500 TypeError ReflectionProperty::getValue(): Argument #1 ($object) must be provided for instance properties

Comment (RemyNovactive) [reply components/MenuManagerBundle/src/lib/Service/DataTransformer/MenuItemsCollectionTransformer.php:41]: @erdnaxelaweb C'est là le problème.

Comment (erdnaxelaweb) [reply components/MenuManagerBundle/src/lib/Service/DataTransformer/MenuItemsCollectionTransformer.php:41]: error fixed by https://github.com/Novactive/Nova-eZPlatform-Bundles/commit/0f2c5bdc30878d5d8eaf515a5eb29fec5ff9fb55

**Pull request merged**